src/protocols/thresholdsig/zilliqa_schnorr.rs

```diff
@@ -66,34 +71,36 @@ impl Keys {
         }
     }
 
-    pub fn phase1_broadcast(&self) -> (KeyGenBroadcastMessage1, BigInt) {
+    pub fn phase1_broadcast(&self) -> (KeyGenBroadcastMessage1, KeyGenMessage2) {
         let blind_factor = BigInt::sample(SECURITY);
         let com = HashCommitment::create_commitment_with_user_defined_randomness(
             &self.y_i.bytes_compressed_to_big_int(),
             &blind_factor,
         );
         let bcm1 = KeyGenBroadcastMessage1 { com };
-        (bcm1, blind_factor)
+        let decom1 = KeyGenMessage2 {
+            y_i: self.y_i,
+            blind_factor,
+        };
+        (bcm1, decom1)
     }
 
     pub fn phase1_verify_com_phase2_distribute(
         &self,
         params: &Parameters,
-        blind_vec: &Vec<BigInt>,
-        y_vec: &Vec<GE>,
+        decom1_vec: &Vec<KeyGenMessage2>,
         bc1_vec: &Vec<KeyGenBroadcastMessage1>,
         parties: &[usize],
     ) -> Result<(VerifiableSS, Vec<FE>, usize), Error> {
         // test length:
-        assert_eq!(blind_vec.len(), params.share_count);
+        assert_eq!(decom1_vec.len(), params.share_count);
         assert_eq!(bc1_vec.len(), params.share_count);
-        assert_eq!(y_vec.len(), params.share_count);
         // test decommitments
         let correct_key_correct_decom_all = (0..bc1_vec.len())
             .map(|i| {
                 HashCommitment::create_commitment_with_user_defined_randomness(
-                    &y_vec[i].bytes_compressed_to_big_int(),
-                    &blind_vec[i],
+                    &decom1_vec[i].y_i.bytes_compressed_to_big_int(),
+                    &decom1_vec[i].blind_factor,
```

Comment:
I think it is better that this function (`phase1_verify_com_phase2_distribute`) will calculate and return `y_sum`.
Otherwise, this is necessarily code that would be written by the consuming application, and it's preferable to be calculated after gathering all the decommitments and verifying them, so I think it's a perfect place to do so.